### DISCUSSION TITLE
Upgrade from Next 11 to Next 13 build error troubleshooting

### DISCUSSION THREAD

**Discussion by AlexSenchenko** [vercel/next.js#45992] [Help]
Hi,

After upgrade from next 11 to next 13 build error. See below.
Need **to find way how to troubleshoot the build process**.  Not clear how to do it now.
Tried to add console.log in each getStaticProps. getStaticPaths, wrap it into try/catch, but it's not fix the issue.
Vercel logs are not really verbose.
Currently got error like: 

```


building [sport]/game/index
--
15:03:02.210 | building [sport]/game/index
15:03:02.424 | building [sport]/game/index
15:03:02.640 | This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
15:03:02.641 | FetchError: invalid json response body at https://vercel.com/login?next=%2Fdeployments%-am0kai4hq-boost.vercel.app%3Fhost%-am0kai4hq-boost.vercel.app%26redirect%3D1%26section%3D reason: Unexpected token < in JSON at position 0
15:03:02.641 | at /vercel/path0/node_modules/node-fetch/lib/index.js:273:32
15:03:02.641 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
15:03:02.658 | This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
15:03:02.658 | FetchError: invalid json response body at https://vercel.com/login?next=%2Fdeployments%-am0kai4hq-boost.vercel.app%3Fhost%-am0kai4hq-boost.vercel.app%26redirect%3D1%26section%3D reason: Unexpected token < in JSON at position 0
15:03:02.659 | at /vercel/path0/node_modules/node-fetch/lib/index.js:273:32
15:03:02.659 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
15:03:02.809 | This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
15:03:02.809 | FetchError: invalid json response body at https://vercel.com/login?next=%2Fdeployments%-am0kai4hq-boost.vercel.app%3Fhost%-am0kai4hq-boost.vercel.app%26redirect%3D1%26section%3D reason: Unexpected token < in JSON at position 0

```
<img width="1195" alt="Screenshot 2023-02-16 at 15 06 25" src="https://user-images.githubusercontent.com/6973991/219360603-9099cb74-026f-4b87-a4e0-4639231af82f.png">

**Comment by icyJoseph** [comment]
Alright, at first this seems to be caused by some data fetching failure. 

Have you also upgraded Node.js version? What Node.js version did you use before?

Latest Node.js versions (15+) exit the Node process when there's an unhandled rejection, before they'd just print a warning.

With that out of the way, the error root seems to be that you are calling `.json()` on a response which doesn't contain valid JSON. Could you share some code snippets of `[sport]/game/index`?

**Comment by AlexSenchenko** [comment]
@icyJoseph  well we not upgrade Nodejs version yet. We use current Lts - v16.13.0
The code is looking like this:

```

export const getStaticPaths = async () => {
  let paths: any[] = [];
  try {
    const sports = await getSportItems();
    const getGamesForSport = (_sport: any) => getGameItems({ sport:
_sport.alias });
    const games = await Promise.all(sports.map(getGamesForSport));
    paths = games.flat()
      .map((game) => {
        const sportId = game.sport?.alias || 'sport';
        const gameId = generateGameId(game);
        console.log(`/${sportId}/game/${gameId}/`);
        return { params: { sport: sportId, game: gameId } };
      });
  } catch (e) {
    console.log({ e });
  }
  return {
    paths: paths.filter((p: any) => p.sport && p.game),
    fallback: 'blocking'
    // fallback: false
  };
};

export const getStaticProps = wrapper
  .getStaticProps((store) => async ({ params }: any = {}) => {
    console.log('building [sport]/game/[id]]');
    let error = '';
    try {
      const baseUrl = envConfig.NEXT_PUBLIC_VERCEL_URL;
      const gameIdParam = params?.gameId;
      const boostGameId = getBoostGameIdFromSlug(gameIdParam);
      await store.dispatch(setActiveMenu(null));
      // await store.dispatch(loadSports());
      await store.dispatch(getFeedContent({ sportName: '', gameId:
boostGameId }));

      return {
        props: { tweets: [] || null, baseUrl },
        revalidate: 60 // revalidate every minute
      } as any;
    } catch (e) {
      console.log(e);
      error = e?.toString()!;
    }

    return {
      // props: { tweets: null },
      props: { error },
      revalidate: 60 // revalidate every minute
    } as any;
  });

```

**Reply by icyJoseph** [reply]
Alright, but node 16 already does the process.exit on unhandled rejection. 

So we got to find where we have an unhandled rejection.

**Comment by icyJoseph** [comment]
The error you are getting is typical of getting HTML/XML, instead of JSON, but I can't see where that call might be happening.

It seems that most async/await work you've got going on is wrapped by try/catch, `wrapper.getStaticProps`?

I'd also try to log the status code of fetch requests that happen when getting this page, see if there's a 500 coming through.

**Reply by AlexSenchenko** [reply]
`wrapper.getStaticProps` it is pure redux toolkit code/function.

> I'd also try to log the status code of fetch requests that happen when getting this page, see if there's a 500 coming through.

You mean try to patch node-fetch lib?

**Reply by icyJoseph** [reply]
no no, just try to find where you make `fetch` calls,  it is probably like:

```js
const response = await fetch('something')

const data = await res.json()
```

Place a `response.status` log between.

I must say this is a very weird error, like it claims it fails because it gets invalid JSON.

I wonder if the issue is when serializaing your data from getStaticProps to the Page Component? But then why does `node-fetch` show up in the logs, ugh, confusing stuff.

**Reply by lbelyaev** [reply]
@icyJoseph thanks for your input, much appreciated. What looks weird to me is that the fetch in question goes toward Vercel https://vercel.com/deployments/{...our Vercel preview url...}?redirect=1, gets 302 (redirect), gets redirected to Vercel.com/login, fails and then exception is thrown.
We obviously don't call Vercel.com/deployments... at any moment. Maybe this gives some clue?). Thanks in advance

**Reply by icyJoseph** [reply]
Hi,

I see, well let's broaden the scope then:

- How does your app consume/fetch data? Is it calling to itself? Like an API route sitting within the server?
- Have you gotten the same Node.js version locally and in Vercel?
- This doesn't happen locally right?
- Maybe try to disable TELEMETRY, https://nextjs.org/telemetry
- What packages are you using? Are you maybe being phished? (unlikely at this point, but who knows)

**Reply by icyJoseph** [reply]
Additionally, maybe there's an issue with your Vercel account? Would be good to open a ticket with their customer support.

**Reply by lbelyaev** [reply]
I seem to have isolated the issue for just a handful of page components which have similar properties and similar data sources. Investigating further. 
To answer your questions:
- telemetry is disabled
- Node is the same (actually tried both 16x and 18x on Vercel with the same result)
- locally all is good
- we consume mostly from the headless CMS (ContentStack), but this works well in other pages
- packets - nothing too fancy, but yes this is one of the places we're currently looking
Thanks for you patience and advice, @icyJoseph !

**Reply by icyJoseph** [reply]
No worries, this is very interesting. How do you integrate with the CMS? 

It'd be really strange if it is a package doing phishing, but let's wait and see.

**Reply by icyJoseph** [reply]
This doesn't apply to you guys right? https://github.com/vercel/next.js/discussions/44730#discussioncomment-5011318

**Reply by AlexSenchenko** [reply]
It seems to be not. 
Not clear why it's tried to look into `FetchError: invalid json response body at https://vercel.com/login?next=`
What is the URL?

**Reply by icyJoseph** [reply]
I found this: https://github.com/vercel/next.js/pull/13202

So is your app trying to reach for it self using VERCEL_URL?
